analysis/quantifier-comparisons/10-miQC-filtering.Rmd

```diff
@@ -0,0 +1,272 @@
+
+```{r}
+# function to filter sce using miQC 
+miQC_filter <- function(sce, model_type = "linear", posterior_cutoff = 0.75){
+  sce_model <- miQC::mixtureModel(sce, model_type)
+  filtered_sce <- miQC::filterCells(sce, sce_model)
+}
+
+# filter using manual 
+manual_filter <- function(sce, 
+                          mito_cutoff = 20, 
+                          genes_detected_cutoff = 500, 
+                          total_count_cutoff = 1000) {
+  
+  # get vector of cells to keep 
+  cells_keep <- sce$detected > genes_detected_cutoff & 
+    sce$sum > total_count_cutoff &
+    sce$subsets_mito_percent < mito_cutoff
+  
+  # filter sce
+  filtered_sce <- sce[,cells_keep]
+}
+
+manual_filtered_sce <- purrr::map(cr_like_sce, manual_filter)
+miQC_filtered_sce <- purrr::map(cr_like_sce, miQC_filter)
+```
+
+
+```{r}
+# function to grab number of cells from sce
+get_num_cells <- function(sce){
+  num_cells <- dim(sce)[2]
+}
+
+manual_cell_num <- manual_filtered_sce %>%
+  purrr::map(get_num_cells) %>%
+  as.data.frame()
+rownames(manual_cell_num) <- c("manual")
+
+miQC_cell_num <- miQC_filtered_sce %>%
+  purrr::map(get_num_cells) %>%
+  as.data.frame()
+rownames(miQC_cell_num) <- c("miQC")
+
+pre_filter_cell_num <- cr_like_sce %>%
+  purrr::map(get_num_cells) %>%
+  as.data.frame()
+rownames(pre_filter_cell_num) <- c("pre_filtering")
+
+# create dataframe for plotting with number of cells per sce 
+combined_cell_num_df <- dplyr::bind_rows(manual_cell_num,
+                                         miQC_cell_num,
+                                         pre_filter_cell_num) %>%
+  tibble::rownames_to_column("filtering_method") %>%
+  tidyr::pivot_longer(cols = starts_with("SCPCR"),
+                      names_to = "sample", 
+                      values_to = "number_cells")
+```
+
+```{r}
+ggplot(combined_cell_num_df, aes(x = sample, y = number_cells, color = filtering_method)) + 
+  geom_point(size = 3)
+```
+
+
+It looks like manual filtering is mostly removing more cells than miQC. 
+Let's look at how the per cell statistics are altered before and after each filtering method. 
+
+```{r}
+# convert coldata for all sces to dataframe 
+manual_filtered_df <- purrr::map(manual_filtered_sce, coldata_to_df) %>%
+  dplyr::bind_rows(.id = "sample")
+
+miQC_filtered_df <- purrr::map(miQC_filtered_sce, coldata_to_df) %>%
+  dplyr::bind_rows(.id = "sample")
+
+pre_filtered_df <- purrr::map(cr_like_sce, coldata_to_df) %>%
+  dplyr::bind_rows(.id = "sample")
+
+# create a list of named filtered dataframes
+list_df <- list(manual_filtered_df,
+                miQC_filtered_df,
+                pre_filtered_df)
+names(list_df) <- c("manual", "miQC", "pre_filtered")
+
+# one combined dataframe with coldata and filtering method as a column 
+filtered_df <- dplyr::bind_rows(list_df, .id = "filtering_method")
+```
+
+
+```{r}
+ggplot(filtered_df, aes(x = filtering_method, y = sum, fill = filtering_method)) +
+  geom_boxplot() +
```

Comment:
I'm not finding these boxplots particularly informative. A density plots might be better, as they would show the whole distribution overlapped, which might allow some finer detail to be seen, but more generally, I'm just not sure that there is much information here beyond the numbers of cells. None of these are doing much that is dramatically different, aside from different cutoffs, and the scatter plots above already allow a reasonable comparison of where the manual cutoff would have been and some of these statistics.